packages/server/utils/MattermostServerManager.ts

```diff
@@ -0,0 +1,95 @@
+import fetch from 'node-fetch'
+
+// Mattermost is a server-only integration for now, unlike the Slack integration
+
+// We're following a similar manager pattern here should we wish to refactor the
+// Mattermost integration. We might want to do this if we decide to use Mattermost's
+// upcoming "App" framework
+
+const MAX_REQUEST_TIME = 5000
+
+export interface MattermostApiResponse {
+  ok: boolean
+  status: number
+  message?: string
+  error?: string
+}
+
+abstract class MattermostManager {
+  webhookUrl: string
+  abstract fetch
+  headers: any
+
+  constructor(webhookUrl: string) {
+    this.webhookUrl = webhookUrl
+  }
+
+  private fetchWithTimeout = async (url: string, options: RequestInit) => {
+    const controller = new AbortController()
+    const {signal} = controller
+    const timeout = setTimeout(() => {
+      controller.abort()
+    }, MAX_REQUEST_TIME)
+    try {
+      const res = await this.fetch(url, {...options, signal})
+      clearTimeout(timeout)
+      return res
+    } catch (e) {
+      clearTimeout(timeout)
+      return new Error('Mattermost is not responding')
+    }
+  }
+
+  private async post(payload: any) {
+    const res = await this.fetchWithTimeout(this.webhookUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json'
+      },
+      body: JSON.stringify(payload)
+    })
+    if (res instanceof Error) return res
+    if (res.status !== 200) {
+      if (res.status === 400) {
+        // Bad request, Matter supplies a JSON error message
+        const {message: error} = await res.json()
```

Comment:
how do we reproduce a 400 error with a json payload? 
i want to make sure that a 400 error has a `json` body payload & not a `text` payload otherwise this method will throw & we'll need to add a catch

Reply:
ICYMI: https://www.loom.com/share/cf1408c5d759470ba4297405104cb0a2